### PR TITLE
ITNFA-24: National registration failures are now retried in task execution

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java [hunk 1]
```diff
@@ -15,8 +22,39 @@
...
+        registrationService = Context.getRegisteredComponent(
+                "registrationService", RegistrationService.class);
+
+        searchService = Context.getRegisteredComponent(
+                "searchService", SearchService.class);
+    }
+
+    private void retryRegistrationFailures() {
+        for (NationalSynchronizationFailure failure : nationalSynchronizationFailureService.findAllRegistrationFailures()) {
+            nationalSynchronizationFailureService.delete(failure);
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java]: I'm not sure if this is a safe approach. What if a registration failure is deleted and then handling fails?

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/model/NationalSynchronizationFailure.java [hunk 2]
```diff
@@ -31,6 +31,10 @@
     @Column(name = "update_failure")
     private boolean update;
 
+    public NationalSynchronizationFailure() {
+
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/model/NationalSynchronizationFailure.java]: undesired empty space

**Review by pawelcieszko — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java (hunk 1 above)
Comment (pawelcieszko) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java]: shouldn't you check if national server is configured?

Comment (dserkowski) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java]: Should't you extract these operations inner 'for' loop to a new method and make it Transactional

File: api/src/main/resources/moduleApplicationContext.xml [hunk 3]
```diff
@@ -29,6 +29,18 @@
         <property name="contextPath" value="org.openmrs.module.m2sysbiometrics.bioplugin"/>
     </bean>
 
+    <bean id="nationalSynchronizationFailureService"
```
Comment (dserkowski) [inline api/src/main/resources/moduleApplicationContext.xml]: You can name beans by parameter value in @Service annotation

**Review by dserkowski — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java (hunk 1 above)
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java]: @pawelcieszko there's no need to check if the national server is configured. It's checked where needed and if it's not set then synchronization simply fails and stays in the queue.

Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/scheduler/NationalSynchronizationTask.java]: @dserkowski Transactional wouldn't work on private method, however, I'll try to make this whole task transactional.

**Comment by dserkowski** [comment]
@oskarhinc resolve conflicts

**Comment by oskarhinc** [comment]
Due to mess with commits in this pull request, I'm closing this one and I've created a new one here: https://github.com/IsantePlus/openmrs-module-m2sys-biometrics/pull/19

**Pull request closed without merging**